tutorials/grid_coordinates.py

```diff
@@ -0,0 +1,324 @@
+"""
+.. _grid_coordinates:
+
+Gridding coordinates
+====================
+
+Grid coordinates in Verde are used to create points on a regularly spaced grid
+that are then used in the spline method to interpolate between sample data
+points. The grid can be specified either by the number of points in each
+dimension (the *shape*) or by the grid node spacing.
+
+Creating regular grids in Verde is done using the :func:`verde.grid_coordinates`
+function. The function uses :func:`numpy.linspace` and
+:func:`numpy.meshgrid` to create a set of regulary spaced points in both the
+west-east and south-north directions to create a two-dimensional spatial grid.
+To this end, the function will adjust either the region bounds or the spacing
+between grid nodes.
+
+First let's create a region that is 1000 units west-east and 1000 units south-
+north, and we will set an initial spacing to 100 units. This should create a
+square grid with no adjustment of the region, or the spacing.
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+import verde as vd
+from matplotlib.patches import Rectangle
+
+# create the bounds for the region and set the spacing for the grid points
+spacing = 100
+west, east, south, north = 0, 1000, 0, 1000
+region = (west, east, south, north)
+
+# create the grid
+square_region_east, square_region_north = vd.grid_coordinates(
+    region=region, spacing=spacing
+)
+
+########################################################################################
+# We can check the dimensions of the grid to confirm that it is 11x11
+
+print(square_region_east.shape, square_region_north.shape)
+
+########################################################################################
+# Now let's plot the grid to visualize the location of the grid points
+
+plt.figure(figsize=(6, 6))
+currentAxis = plt.gca()
+currentAxis.add_patch(
+    plt.Rectangle((west, south), east, north, fill=None, label="Region Bounds")
+)
+
+plt.vlines(
+    square_region_east[0],
+    ymin=min(square_region_north[0]),
+    ymax=max(square_region_north[-1]),
+    linestyles="dotted",
+    zorder=0,
+)
+plt.hlines(
+    square_region_north[:, 1],
+    xmin=min(square_region_east[0]),
+    xmax=max(square_region_east[-1]),
+    linestyles="dotted",
+    zorder=0,
+)
+
+plt.scatter(
+    square_region_east,
+    square_region_north,
+    label="Square Region Grid Nodes",
+    marker=".",
+    color="black",
+    s=100,
+)
+
+plt.ylim(-50, 1050)
+plt.xlim(-50, 1050)
+plt.xlabel("Easting")
+plt.ylabel("Northing")
+plt.legend(loc="upper center", bbox_to_anchor=(0.5, 1.15))
+plt.show()
+
+########################################################################################
+# How Grid Coordinates works

```

Comment:
I would simplify these three sections ("How Grid Coordinates works", "Create the grid with adjust region" and "Create the grid with adjust spacing") into two: maybe "Adjusting region boundaries when creating the grid" and "Adjusting spacing when creating the grid".

So maybe it could look like this:
```python
# Adjusting region boundaries when creating the grid
# --------------------------------------------------
# Now let's create a region that is 1555 units west-east, and 1250 units south-north
# with a spacing of 500 units. Because the range of the west-east and south-north
# boundaries are not multiples of 500, we must choose to change either:
#   - the boundaries of the region in order to fit the spacing, or
#   - the spacing in order to fit the region boundaries.
#
# We could tell :func:`verde.grid_coordinates` to adjust the region boundaries by
# passing `adjust="region"`.

region_east, region_north = vd.grid_coordinates(
    region=region, spacing=spacing, adjust="region"
)
print(region_east.shape, region_north.shape)

...
```

```python
# Adjusting spacing when creating the grid
# ----------------------------------------
# Now let's adjust the spacing of the grid points by passing `adjust="spacing"` to
# :func:`verde.grid_coordinates`.

spacing_east, spacing_north = vd.grid_coordinates(
    region=region, spacing=500, adjust="spacing"
)
print(spacing_east.shape, spacing_north.shape)

...
```

What do you think?

Reply:
I think that three sections with your proposed headers make a lot of sense, and have a solid flow to them. I have changed them in the tutorial